srcs/parse/tokenize.c

```diff
@@ -0,0 +1,73 @@
+#include "minishell.h"
+#include "typedefs.h"
+#include "xlibft.h"
+
+int	count_token_num(char *str)
+{
+	size_t	i;
+	int		ans;
+
+	i = 0;
+	ans = 1;
+	while (str[i] != '\0')
+	{
+		if (str[i] == ' ')
+			ans += 1;
+		if (str[i] == '|')
+			ans += 2;
+		i++;
+	}
+	return (ans + 1);
+}
+
+t_token	*initialize_t_token(char *str)
+{
+	t_token	*ans;
+
+	ans = (t_token *)ft_xcalloc(sizeof(t_token), 1);
+	ans->token = (int *)ft_xcalloc(sizeof(int), count_token_num(str));
+	ans->word = (char **)ft_xcalloc(sizeof(char *), count_token_num(str));
+	return (ans);
+}
+
+bool	is_special_char(char c)
+{
+	if (c == ' ' || c == '\0' || c == '|' || c == '&')
+		return (true);
+	return (false);
```

Comment:
```suggestion
	return (c == ' ' || c == '\0' || c == '|' || c == '&');
```

評価式をそのままreturnしても良いかも？